### ISSUE TITLE
Ability to manually add a team member to the team rather than requesting signups.

### ISSUE THREAD

**Issue by hakt0-r** [timezoneio/timezoneio#9]
As the title suggests, rather than requesting People sign up to get their timezones incorporated into a team, can you allow the team manager to add employees with different timezone without the need for them to sign up. This way we can use the site more frequently and its more useful as I can see various timezones in one glance that mean something to me rather than being employee focused.

**Comment by djfarrelly** [comment]
Hey @hakt0-r, I'm curious about this one, currently you can add a user to the team and they can ignore the signup. Would you rather skip that invite email being sent out to the user and request they sign up?

**Comment by hakt0-r** [comment]
Yes, consider this example, I am a consultant and I employ people for various jobs, some of them are once off and the others I'd like to make them part of the team. So those guys who'll be working regularly with me, I'd prefer they sign up so that they can know when a specific project's team members are online, while the temporary ones I'd rather just know when to talk to them or not, like placeholders.

**Comment by hakt0-r** [comment]
this example can be extended to allow for groups being created for a company/projects too. A group may be called developers and the other clients and I can add members to both. The key here is if I share the link to the project with someone I'd like them to sign up too for this.

**Comment by djfarrelly** [comment]
Thanks for the thoughts @hakt0-r! This triggers a couple of ideas that I'd like your input on if you have time:

- When you add a person onto your team, you have 2 buttons "Add and invite with email" or "Add manually." - I held off on this initially thinking the current model would help drive signups, I also thought the two button approach might add friction to create a team. After your input and some input from others, the 2 button idea is starting to feel good again.
- Having a **"My Team"** which is only visible to you where you can add people or simply places without the need for an invite. This would be private to you and you can add a person temporarily like "Joe Client in Vancouver" or a location like "London" - Public teams would be separate and be encourage to share them with others and have other update their location. 
- For your second item, in essence you'd like to allow others to manage your team as well as allow people with the link to add themselves?

**Comment by hakt0-r** [comment]
Hi Dan,

> Having a "My Team" which is only visible to you where you can add people or simply places without the need for an invite. This would be private to you and you can add a person temporarily like "Joe Client in Vancouver" or a location like "London" - Public teams would be separate and be encourage to share them with others and have other update their location. 
Perfect! Now you see exactly what I want. if I add people without allowing them to sign up, I’d like to have that information private, e.g. client list, etc… But, like you said, if I invite people in my team(s) then they can see other members of that team only. 

The way this will work the best is to have a base company -> Teams (or groups or whatever you call them) -> Team members. At the end of the day, if I look at the base company, I can see everyone, both manually added and invited members arranged (perhaps) in columns for teams.

> For your second item, in essence you'd like to allow others to manage your team as well as allow people with the link to add themselves?
This will work only if that team manager cannot see outside that team, i.e. they cannot traverse vertically up to the base company and then down to other teams. They must be confined to teams (or groups). 

Another way to look at this would also be that a developer (or team member) can belong to multiple teams too.

Thank you for listening and considering these changes. It will certainly make this site a lot more valuable tool for me with these changes.

Regards,

Vik Shah
Director
Keys2Solutions

**Comment by djfarrelly** [comment]
Thanks for that input on the idea of a "My Team" or a greater "umbrella team." I think that could be a cool concept. Here's what I'm thinking:

1. Each user has a default "My Team" umbrella team which displays all people, places they've manually added and the people from all the teams that they're on.
2. A team can add sub-groups 1 level deep; i.e. "Company X" has 3 groups "Developers" "Support" and "Account managers"

I think these would be great and are things I've been thinking about with the size that the Buffer team has been growing to (http://timezone.io/team/buffer).

**Comment by hakt0-r** [comment]
Hi Dan,

> Each user has a default "My Team" umbrella team which displays all people, places they've manually added and the people from all the teams that they're on.
Yup, and if you look at this from the team members dashboard, you can pivot this concept too for them. They’ll see which teams they are assigned to as well as any manual/invited contacts they have created from their perspective. They may not call it a company but I think you get the idea. :-)

> A team can add sub-groups 1 level deep; i.e. "Company X" has 3 groups "Developers" "Support" and "Account managers"

Yup! I think from my simpleton perspective going 1 level deep should be enough. Perhaps in the future you could setup the teams table to have a many=to=many relationship and that way you can create unlimited sub-groups, the only issue being the query joins will be infinitely complicated once you peer more than a few inherited levels deep. I’d recommend 2 tiers at the most (for the future).

thank you once again.

Regards,

Vik Shah
Director
Keys2Solutions

**Comment by tzsasaki** [comment]
hello hello, I would also love the ability to 1) manually add people 2) create subgroups, with the former being much more interesting that the latter. 

My team is in 2-3 time zones max, so the real value is in being able to additionally see clients and contributors (which I would add manually) on the same view. 

Great work @djfarrelly looking forward to how the app progresses!

**Comment by nsudds** [comment]
Looking forward to this feature, I'd like to build these views for my own benefit but not necessarily invite others to view it --- I also think it would be interesting to add a private "group" for clients and be able to manually add them to see your client base across time zones.  If there was a way to import their avatar/photo via some integration that would make it easier.  i.e. Adding them from linkedin/facebook/google/gravatar  (by putting in their email but not subscribing them)  

In my mind, I'd like to explore the idea of the added bonus of not just keeping your team organized but clients front of mind with their avatars and time zones so you know the best time to reach out to them.

**Comment by djfarrelly** [comment]
Great points @nsudds, I've received some recent feedback emails asking for the "private team" feature.

The way I was envisioning that is that when you sign up your get your own private team by default, being able to add up to X locations/companies/people or you can choose to add your team by email invite or via Slack.

Re: the avatars, that's probably one of the more annoying things you have to do when adding a person to your team. When you mean "adding them by linkedin/facebook...", do you mean having like a search of your Facebook friends or pages to pull in the photo or something else entirely?
